### PR TITLE
Feature/better support for parameterized test

### PR DESCRIPTION
**Pull request by john7doe**
add support for JParam and normal Parameterized tests

### CODE REVIEW

**Comment by john7doe** [comment]
context: https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/23721

File: src/main/java/com/microsoft/appcenter/event/EventReporter.java [hunk 1]
```diff
@@ -17,20 +17,22 @@
     private Description lastDescription = null;
     private int labelCounter = 0;
     private int screenshotCounter = 0;
+
+    // Parameters will always be in "[]": https://github.com/junit-team/junit4/blob/7111b9621997f6c660687b8ac04003398343ee3a/src/main/java/org/junit/runners/Parameterized.java#L484
     private final Pattern paramsFinder = Pattern.compile("\\[(.+?)\\]");
```
Comment (kamstrup) [inline src/main/java/com/microsoft/appcenter/event/EventReporter.java:22]: I know you did not touch this regex, but it looks suspicious... Firstly I guess '.+?' is equivalent to '.*'? Second it matches on the entire string "[oneThing[onsubThing]] [somethingElse]"

File: src/test/java/com/microsoft/appcenter/espresso/ParameterizedNamedUniqIdsTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public ReportHelper reportHelper = new ReportHelper(testableEventReporter);
+
+    @Parameterized.Parameters(name = "{index}: foo {0}")
+    public static Iterable<Integer> data() {
+        return Arrays.asList(7, 9, 13);
+    }
+
+    private int input;
+
+    public ParameterizedNamedUniqIdsTest(int input) {
```
Comment (kamstrup) [inline src/test/java/com/microsoft/appcenter/espresso/ParameterizedNamedUniqIdsTest.java:39]: Ha! Gotcha, found 1 place with C# naming ;-D

**Review by kamstrup — APPROVED** [review]
Great stuff Simon!

Just 1 instance of C# naming. I am willing to let that one slip since it's a lone one ;-)

And then there is a suspicious regex, but you didn't change that code, so you're off the hook there I guess :-)

File: src/test/java/com/microsoft/appcenter/espresso/ParameterizedNamedUniqIdsTest.java (hunk 2 above)
Comment (john7doe) [reply src/test/java/com/microsoft/appcenter/espresso/ParameterizedNamedUniqIdsTest.java:39]: its a constructor, so no  ;-)

**Pull request merged**